scipy/optimize/_linprog_highs.py

```diff
@@ -0,0 +1,186 @@
+    parallel : bool
+        Set to ``True`` to enable parallelization; default ``False``.
+    unknown_options : dict
+        Optional arguments not used by this particular solver. If
+        `unknown_options` is non-empty, a warning is issued listing all
+        unused options.
+
+    Returns
+    -------
+    sol : dict
+        A dictionary consisting of the fields:
+
+            x : 1D array
+                The values of the decision variables that minimizes the
+                objective function while satisfying the constraints.
+            fun : float
+                The optimal value of the objective function ``c @ x``.
+            slack : 1D array
+                The (nominally positive) values of the slack variables,
+                ``b_ub - A_ub @ x``.
+            con : 1D array
+                The (nominally zero) residuals of the equality constraints,
+                ``b_eq - A_eq @ x``.
+            success : bool
+                ``True`` when the algorithm succeeds in finding an optimal
+                solution.
+            status : int
+                An integer representing the exit status of the algorithm.
+
+                ``0`` : Optimization terminated successfully.
+
+                ``1`` : Iteration limit reached.
+
+                ``2`` : Problem appears to be infeasible.
+
+                ``3`` : Problem appears to be unbounded.
+
+                ``4`` : Numerical difficulties encountered.
+
+            nit : int
+                The total number of iterations performed in all phases.
+            message : str
+                A string descriptor of the exit status of the algorithm.
+
+    """
+
+    _check_unknown_options(unknown_options)
+
+    messages = ["Optimization terminated successfully.",
+                "Iteration limit reached.",
+                "The problem appears infeasible, as the phase one auxiliary "
+                "problem terminated successfully with a residual of {0:.1e}, "
+                "greater than the tolerance {1} required for the solution to "
+                "be considered feasible. Consider increasing the tolerance to "
+                "be greater than {0:.1e}. If this tolerance is unnaceptably "
+                "large, the problem is likely infeasible.",
+                "The problem is unbounded, as the simplex algorithm found "
+                "a basic feasible solution from which there is a direction "
+                "with negative reduced cost in which all decision variables "
+                "increase.",
+                "Numerical difficulties encountered; consider trying "
+                "method='interior-point'.",
+                ]
+
+    c, A_ub, b_ub, A_eq, b_eq, bounds, x0 = lp
+    lb, ub = bounds.T                       # separate bounds, TODO: C-cntgs
+    # highs_wrapper solves LHS <= A*x <= RHS, not equality constraints
+    lhs_ub = None                           # LHS of UB constraints is -inf
```

Comment:
This needs to be an array of `-np.inf` to concatenate, but I was having issues with that.

Reply:
Passing in `lhs_ub=None` creates an array `-inf` automatically, so no need

Reply:
Passing `lhs=None` into the wrapper works, but I need to concatenate `lhs_ub` and `lhs_eq`

Reply:
Ah, I see.  What actually happens on HiGHS side is you make it a really large number and then HiGHS recognizes it as infinity (or small for -infinity).  So if you do:

`lhs = -1e-20np.ones((n,), dtype='double')`

I think I can expose the actual constant they use for threshold for infinities, I make an issue to remind myself ([#2](https://github.com/mckib2/HiGHS/issues/2))